### PR TITLE
Only autoenqueue podcasts with a valid or empty type

### PR DESCRIPTION
**Pull request by dennisschagt**
Fixes https://github.com/newsboat/newsboat/issues/2367

### CODE REVIEW

File: src/queuemanager.cpp [hunk 1]
```diff
@@ -122,10 +122,19 @@ EnqueueResult QueueManager::autoenqueue(std::shared_ptr<RssFeed> feed)
 			continue;
 		}
 
+		const auto enclosure_type = item->enclosure_type();
+		const auto enclosure_url = item->enclosure_url();
+
+		if (!enclosure_type.empty() && !utils::is_valid_podcast_type(enclosure_type)) {
+			LOG(Level::DEBUG, "QueueManager::autoenqueue: Skipping enclosure with url `%s'"
+				" becausoe of invalid podcast type `%s'", enclosure_url, enclosure_type);
```
Comment (der-lyse) [inline src/queuemanager.cpp:130]:
```suggestion
				" because of invalid podcast type `%s'", enclosure_url, enclosure_type);
```

Comment (dennisschagt) [reply src/queuemanager.cpp]: Oops, fixed

**Review by Minoru — CHANGES_REQUESTED** [review]
Sorry I'm so late with the review!

I agree with the new `if` condition, but I think it should be in `RssParser`. In #2218 individual parsers (RSS 1.0, Atom etc.) learned to extract *all* the enclosures, and `RssParser` was made responsible for picking the "best" enclosure. I think we should just to modify the algorithm there.

**Comment by dennisschagt** [comment]
No problem, thank you for the review anyway!

> In https://github.com/newsboat/newsboat/pull/2218 individual parsers (RSS 1.0, Atom etc.) learned to extract all the enclosures, and RssParser was made responsible for picking the "best" enclosure. I think we should just to modify the algorithm there.

I prefer to keep the decision of which items to (auto-)enqueue in the queuemanager.
My idea for enclosures and other media attachments is as follows:
- Support multiple attachments (both enclosures and `media:content` elements, removing the need to pick the "best" enclosure)
- Enqueue only items with audio/video/unknown mime-type (specifically, don't enqueue images => check added in this PR)
- Show image attachments near the top of the itemview (between the headers and the actual text, rendering them similar to links specified in item's HTML content)
- Related idea: https://github.com/newsboat/newsboat/issues/2305#issuecomment-1493412650

To be able to show images, we should not already filter them out in the `RssParser`.

Does that make sense to you / do you agree?

**Review by Minoru — APPROVED** [review]
Yup, makes total sense! I failed to consider how we could evolve this.

**Pull request merged**